addon/batcher.ts

```diff
@@ -1,19 +1,57 @@
-import { buildWaiter, ITestWaiter, Token } from 'ember-test-waiters';
+import { ITestWaiter, Token, buildWaiter } from 'ember-test-waiters';
+
+import { DEBUG } from '@glimmer/env';
 
 type MaybeRequestAnimationFrame = (callback: FrameRequestCallback | Function) => number;
 type DomOperation = [Token, () => void];
 
+const IS_BROWSER = typeof window === 'object' && typeof document === 'object';
 const SCHEDULE_MACROTASK = (callback: Function) => setTimeout(callback);
-const scheduleFn: MaybeRequestAnimationFrame =
-  typeof window === 'object' && typeof window.requestAnimationFrame === 'function'
-    ? window.requestAnimationFrame
-    : SCHEDULE_MACROTASK;
 const readDOMWaiter: ITestWaiter = buildWaiter('ember-batcher: readDOM');
 const mutateDOMWaiter: ITestWaiter = buildWaiter('ember-batcher: mutateDOM');
 
 const reads: Array<DomOperation> = [];
 const mutations: Array<DomOperation> = [];
+let visibilityChange: Function = () => {};
 let running: boolean = false;
+let scheduleFnExecuted: boolean = false;
+
+const rafRace = (callback: Function) => {
+  setTimeout(() => {
+    if (!scheduleFnExecuted) {
+      callback();
+    }
+  }, 20);
```

Comment:
I think you can make this tighter, 0ms

Reply:
I think the idea here was to give the execution 1 frame (1s/60fps = 16ms) of breathing room. This will hopefully ensure any pending renders are complete. Thoughts?

Reply:
Ya, just to confirm, doing:

```js
setTimeout(() => console.log('setTimeout'), 0);
requestAnimationFrame(() => console.log('rAF'));
```

Results in different output depending on browser. In Chrome, the rAF is always fired first, but in Safari the setTimeout is. 

Using the 20ms delay is more in line with the intent: that we should only fire in a setTimeout if we do not think the rAF method is going to work.

Reply:
I'm going to go ahead and merge this. If you feel strongly about revisiting, @stefanpenner, I'm happy to revisit!